lib/HLSL/ComputeViewIdStateBuilder.cpp

```diff
@@ -454,35 +458,44 @@ void DxilViewIdStateBuilder::AnalyzeFunctions(EntryInfo &Entry) {
 }
 
 void DxilViewIdStateBuilder::CollectValuesContributingToOutputs(
-    EntryInfo &Entry) {
+    EntryInfo &Entry, bool IsForPatchConstantOrPrimitive) {
   for (auto *CI : Entry.Outputs) { // CI = call instruction
     DxilSignature *pDxilSig = nullptr;
     Value *pContributingValue = nullptr;
     unsigned id = (unsigned)-1;
     int startRow = Semantic::kUndefinedRow, endRow = Semantic::kUndefinedRow;
     unsigned col = (unsigned)-1;
     if (DxilInst_StoreOutput SO = DxilInst_StoreOutput(CI)) {
+      if (IsForPatchConstantOrPrimitive)
+        continue;
       pDxilSig = &m_pModule->GetOutputSignature();
       pContributingValue = SO.get_value();
       GetUnsignedVal(SO.get_outputSigId(), &id);
       GetUnsignedVal(SO.get_colIndex(), &col);
       GetUnsignedVal(SO.get_rowIndex(), (uint32_t *)&startRow);
     } else if (DxilInst_StoreVertexOutput SVO =
                    DxilInst_StoreVertexOutput(CI)) {
+      if (IsForPatchConstantOrPrimitive)
+        continue;
       pDxilSig = &m_pModule->GetOutputSignature();
       pContributingValue = SVO.get_value();
       GetUnsignedVal(SVO.get_outputSigId(), &id);
       GetUnsignedVal(SVO.get_colIndex(), &col);
       GetUnsignedVal(SVO.get_rowIndex(), (uint32_t *)&startRow);
     } else if (DxilInst_StorePrimitiveOutput SPO =
                    DxilInst_StorePrimitiveOutput(CI)) {
+      if (!IsForPatchConstantOrPrimitive)
+        continue;
       pDxilSig = &m_pModule->GetPatchConstOrPrimSignature();
       pContributingValue = SPO.get_value();
       GetUnsignedVal(SPO.get_outputSigId(), &id);
       GetUnsignedVal(SPO.get_colIndex(), &col);
       GetUnsignedVal(SPO.get_rowIndex(), (uint32_t *)&startRow);
+      IsForPatchConstantOrPrimitive = true;
     } else if (DxilInst_StorePatchConstant SPC =
                    DxilInst_StorePatchConstant(CI)) {
+      if (!IsForPatchConstantOrPrimitive)
```

Comment:
I see that `IsForPatchConstantOrPrimitive` is set to true if it is a `DxilInst_StorePrimitiveOutput`.  Why doesn't it also get set to true for `DxilInst_StorePatchConstant`?

Reply:
Or alternatively, why does it get set?  It seems that it puts this function into a different mode. Does this mean that the order of `Entry.Outputs` is significant here?

Reply:
We have 2 modes in CollectValuesContributingToOutputs.
And we'll call CollectValuesContributingToOutputs with different FuncInfo, one for check normal output and vertex output (outputs saved in OutputSignature), one for check patch constant output and primitive output (outputs saved in PatchConstOrPrimSignature).
The order doesn't important here, we just scan all the outputs collected in early pass.

Without this change, we put vertex output and primitive output in one table for mesh shader, which is wrong.
Patch constant doesn't have this issue because patch constant has its own function.

But mesh shader only has one entry, so when in vertex mode, we only scan vertex output, in primitive mode, we only scan primitive.

Reply:
I understand that there are two modes. What I don't understand is why it changes mode when a `DxilInst_StorePrimitiveOutput` is encountered, on line 494. That very much makes it seem like the order of outputs matters, since after the first `DxilInst_StorePrimitiveOutput` is encountered all remaining outputs will be ignored.

Reply:
Got it. Missed dead code when update the code :(
Fixed.

Reply:
If the tests weren't capturing this, it strongly suggests to me that the testing coverage for this change needs to be improved.

Reply:
It is impossible to capture it, since it is dead code :(
If the code can reach line 494, the `IsForPatchConstantOrPrimitive` must be true.
So `IsForPatchConstantOrPrimitive = true;` will not change anything.

Reply:
Ah, I didn't notice that the various tests on `IsForPatchConstantOrPrimitive` weren't all the same.